### PR TITLE
fix(pubsub): check both client errors on Close

### PR DESCRIPTION
**Pull request by hongalex**
In #2624, we split the publisher and subscriber client's connection pool. Thus, we need to check both errors returned, rather than just the first one.

### CODE REVIEW

**Review by shollyman — APPROVED** [review]
Approving, but please see the comment about close semantics and address if necessary.

**Review by codyoss — APPROVED** [review]

File: pubsub/pubsub.go [hunk 1]
```diff
@@ -78,14 +78,11 @@ func NewClient(ctx context.Context, projectID string, opts ...option.ClientOptio
...
 	pubc, err := vkit.NewPublisherClient(ctx, o...)
 	if err != nil {
-		return nil, fmt.Errorf("pubsub: %v", err)
+		return nil, fmt.Errorf("pubsub(publisher): %v", err)
 	}
 	subc, err := vkit.NewSubscriberClient(ctx, o...)
 	if err != nil {
-		// Should never happen, since we are passing in the connection.
-		// If it does, we cannot close, because the user may have passed in their
-		// own connection originally.
```
Comment (tbpg) [inline pubsub/pubsub.go:87]: Is this comment no longer true?

File: pubsub/streaming_pull_test.go [hunk 2]
```diff
@@ -315,10 +315,12 @@ func TestStreamingPull_ClosedClient(t *testing.T) {
...
 	time.Sleep(100 * time.Millisecond)
 
-	err := client.Close()
-	if err != nil {
-		t.Fatal(err)
-	}
+	// Intentionally don't check the returned err here. In the fake,
+	// closing either the publisher/subscriber client will cause the
+	// server to clean up resources, which is different than in the
+	// live service. With the fake, client.Close() will return an
```
Comment (tbpg) [inline pubsub/streaming_pull_test.go:321]: Nit: s/an/a/

File: pubsub/pubsub.go [hunk 3]
```diff
@@ -101,10 +98,13 @@ func NewClient(ctx context.Context, projectID string, opts ...option.ClientOptio
 // If the client is available for the lifetime of the program, then Close need not be
 // called at exit.
 func (c *Client) Close() error {
-	// Return the first error, because the first call closes the connection.
-	err := c.pubc.Close()
-	_ = c.subc.Close()
-	return err
+	if err := c.pubc.Close(); err != nil {
```
[the remark this answers is not in the document]
Comment (hongalex) [reply pubsub/pubsub.go]: Agreed. Restructured to resolve this.

File: pubsub/pubsub.go (hunk 1 above)
Comment (hongalex) [reply pubsub/pubsub.go:87]: Correct. Before, the publisher connection was passed in to the subscriber connection, but this was separated in #2624.

**Comment by xmlking** [comment]
@hongalex I was debugging why I am getting `"pubsub publisher closing error: rpc error: code = Canceled desc = grpc: the client connection is closing`  error when called `client.Close()` 
After upgrading to `cloud.google.com/go/pubsub v1.8.0`  I started see this error. I never got this error in previous pubsub versions. 

I accidentally discovered there is a type in the error message at:  
https://github.com/googleapis/google-cloud-go/blob/master/pubsub/pubsub.go#L100-L109

this should be:
```go
func (c *Client) Close() error {
	pubErr := c.pubc.Close()
	subErr := c.subc.Close()
	if pubErr != nil {
                 return fmt.Errorf("pubsub publisher closing error: %v", pubErr)
	}
	if subErr != nil {
		return fmt.Errorf("pubsub subscriber closing error: %v", subErr)
	}
	return nil
}
```

**Comment by xmlking** [comment]
Should we detect `PubSub emulator`  environment and automatically  suppress  `the client connection is closing` error ?
otherwise this leads to confusion  to users

**Comment by hongalex** [comment]
Hey @xmlking, this error does occur with the emulator and the `pstest` fake server. I'll address this in a separate PR so we can suppress "the client connection is closing" errors. Also, thanks for catching the typo.

Btw, I noticed in the linked issue, [there's a comment](https://github.com/xmlking/toolkit/blob/804d63dab872317336ef2a2f6f2cca21e2dba6cb/broker/pubsub/default.go#L108) that says 
> Live pubsub server will throw this error

Though I think the opposite is true (live pub/sub service will *not* throw this error).

**Comment by xmlking** [comment]
thanks @hongalex . fixed my type

**Pull request merged**